From 2b5b425e2347bef872fee3e9119b06ccb0856ba0 Mon Sep 17 00:00:00 2001
From: Haibo Chen <haibo.chen@nxp.com>
Date: Fri, 8 Jan 2016 20:15:05 +0800
Subject: [PATCH] MLK-12140 ARM: dts: imx7d-sdb-epdc: add EPDC touch support
 for imx7d-sdb

EPDC board has touch screen, this patch add touch support on imx7d-sdb TO1.1
board, due to the touch pin conflict with UART5 and GPMI, so disable UART5 in
imx7d-sdb-epdc.dtsi. For GPMI, it default disabled in imx7d-sdb.dts, so do not
impact touch.

Signed-off-by: Haibo Chen <haibo.chen@nxp.com>
(cherry picked from commit f9936728ca87834662a0fcb574d362a299ed337e)
---
 arch/arm/boot/dts/imx7d-sdb-epdc.dtsi | 19 +++++++++++++++++++
 arch/arm/boot/dts/imx7d-sdb.dts       |  8 ++++++++
 2 files changed, 27 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi b/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi
index cbc7ca1..a04aca7 100644
--- a/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi
+++ b/arch/arm/boot/dts/imx7d-sdb-epdc.dtsi
@@ -33,3 +33,22 @@
 &sim1 {
 	status = "disabled";
 };
+
+&uart5 {
+	status = "disabled";
+};
+
+&i2c3 {
+	elan@10 {
+		pinctrl-names = "default";
+		pinctrl-0 = <&pinctrl_epdc_elan_touch>;
+		compatible = "elan,elan-touch";
+		reg = <0x10>;
+		interrupt-parent = <&gpio6>;
+		interrupts = <12 IRQ_TYPE_EDGE_FALLING>;
+		gpio_elan_cs = <&gpio6 13 0>;
+		gpio_elan_rst = <&gpio6 15 0>;
+		gpio_intr = <&gpio6 12 0>;
+		status = "okay";
+	};
+};
diff --git a/arch/arm/boot/dts/imx7d-sdb.dts b/arch/arm/boot/dts/imx7d-sdb.dts
index 55cdbd7..a206341 100644
--- a/arch/arm/boot/dts/imx7d-sdb.dts
+++ b/arch/arm/boot/dts/imx7d-sdb.dts
@@ -556,6 +556,14 @@
 			>;
 		};
 
+		pinctrl_epdc_elan_touch: epdc_elan_touch_grp {
+			fsl,pins = <
+				MX7D_PAD_SAI1_RX_DATA__GPIO6_IO12 0x59
+				MX7D_PAD_SAI1_TX_DATA__GPIO6_IO15 0x1b
+				MX7D_PAD_SAI1_TX_BCLK__GPIO6_IO13 0x80000000
+			>;
+		};
+
 		pinctrl_ecspi3_cs: ecspi3_cs_grp {
 			fsl,pins = <
 				MX7D_PAD_SD2_CD_B__GPIO5_IO9	0x80000000
-- 
1.8.0

